File: mentecart/backend/src/models/service.model.ts
import mongoose from "mongoose";




const serviceSchema = new mongoose.Schema(
    {
        title:{
            type:String,
            required:true,
            trim: true,
        },

        descreption:{
            type: String,
            required:true,
        },

        price:{
            type:Number,
            required:true,
},

        duration:{

        type:Number,
        required:true,
},

        category:{
            type:String,
            required: true,
        },

        image:{
            type: String,
            required: true,
        },

        capacityPerSlot: {
            type: Number,
            required: true,
            default:1,
        },
    },
    {
        timestamps:true,
    }
);

export const Service = mongoose.model(
    "Service",
    serviceSchema
);

File: mentecart/mobile/lib/features/services/data/models/service_model.dart
class ServiceModel {
  final String id;
  final String title;
  final String description;
  final int price;
  final String category;
  final String image;

  ServiceModel({
    required this.id,
    required this.title,
    required this.description,
    required this.price,
    required this.category,
    required this.image,
  });

  factory ServiceModel.fromJson(
    Map<String, dynamic> json,
  ) {
    return ServiceModel(
      id: json['_id'],
      title: json['title'],
      description: json['description'],
      price: json['price'],
      category: json['category'],
      image: json['image'],
    );
  }
}